### PR TITLE
last error header changes-1

### PR DESCRIPTION
**Pull request by sarab96**
last error header changes-1
Adding PID field as well

### CODE REVIEW

**Review by larry9523 — COMMENTED** [review]
Since there is no code use these structure, so it is not clear to me what is the plan. In zocl, there is similar data structure which are used by kernel driver only. So I am wondering if xclerr.h is the right place for them. Should they be put to a common header that shared by zocl and xocl/xclmgmt in kernel space only?

File: src/runtime_src/core/include/xclerr.h [hunk 1]
```diff
@@ -88,4 +92,22 @@ struct xclErrorStatus {
...
+	xrtErrorTime	ts;		/* 64 bits; timestamp */
+	unsigned        pid;            /* 32 bits; pid associated with error, if available */
+};
+
+struct xocl_errors {
+	int		num_err;	/* number of errors recorded */
+	int		cap;		/* capacity of current error array */
+	struct xclErrorLast *errors;	/* error array pointer */
+};
+
```
Comment (stsoe) [inline src/runtime_src/core/include/xclerr.h:112]: What is the intended end-use of these structs?  I am concerned that these structs are added to an exported public header file and that we will have application code dependencies on struct and their sizes.

Please provide the bigger picture here in terms of how these structs will be used and by whom.

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/include/xclerr.h (hunk 1 above)
Comment (sarab96) [reply src/runtime_src/core/include/xclerr.h]: updated based on discussion

**Comment by sarab96** [comment]
updated based on discussion

**Review by stsoe — APPROVED** [review]

File: src/runtime_src/core/include/xclerr_int.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+struct xclErrorLast {
+	xrtErrorCode	zer_err_code;	/* 64 bits; XRT error code */
+	xrtErrorTime	zer_ts;		/* 64 bits; timestamp */
+	unsigned        pid;            /* 32 bits; pid associated with error, if available */
+};
+
+typedef struct xocl_errors {
+	int		num_err;	/* number of errors recorded */
+	int		cap;		/* capacity of current error array */
+	struct xclErrorLast *errors;	/* error array pointer */
```
Comment (larry9523) [inline src/runtime_src/core/include/xclerr_int.h:67]: This struct is going to be interpreted from a sysfs node binary format, right? I assume it should be a contiguous chunk of  memory. Then struct xclErrorLast errors[1] seems better than a pointer, right?

**Review by larry9523 — COMMENTED** [review]
see inline comment

**Review by larry9523 — APPROVED** [review]

File: src/runtime_src/core/include/xclerr_int.h (hunk 2 above)
Comment (sarab96) [reply src/runtime_src/core/include/xclerr_int.h]: correct. updated

**Comment by sarab96** [comment]
updated
> see inline comment

**Review by maxzhen — APPROVED** [review]

**Pull request merged**